ledger/error/src/main/scala/com/daml/error/ErrorCode.scala

```diff
@@ -50,6 +51,25 @@ abstract class ErrorCode(val id: String, val category: ErrorCategory)(implicit
 
   implicit val code: ErrorCode = this
 
+  /** @return whether the supplied exception matches this error code.
+    */
+  def matches(e: StatusRuntimeException): Boolean = {
+    val matchesErrorCodeId = from(e).exists {
+      case ErrorInfoDetail(errorCodeId, _) => errorCodeId == this.id
+      case _ => false
+    }
+    val matchesMessagePrefix = e.getStatus.getDescription.startsWith(
+      LedgerApiErrors.AdminServices.UserAlreadyExists.id
```

Comment:
This doesn't seem applicable in all circumstances

Reply:
Fixing